### PR TITLE
Update rust binding

### PR DESCRIPTION
**Pull request by cryptoAtwill**
Rust binding in dev is outdated. Adding auto commit in github action.

### CODE REVIEW

**Review by adlrocha — APPROVED** [review]
LGTM. Thanks!

Just one quick concern, are we sure that the Rust compilation is deterministic? I remember we were having some issues with bindings changing between compilation even if the code hadn't changed. I am asking because this could lead to a lot of noise with bindings changing in every PR.

**Comment by cryptoAtwill** [comment]
@adlrocha from my past experience, without reading the compilation source code or ethers, I cannot say the binding is deterministic. But this way at least we won't have to worry about binding up to date.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -12,6 +12,7 @@ jobs:
       - name: Checkout Repository
         uses: actions/checkout@v3
         with:
+          ref: ${{ github.event.pull_request.head.ref }}
```
Comment (aakoshh) [inline .github/workflows/test.yml:15]: Why is this needed?

Comment (cryptoAtwill) [reply .github/workflows/test.yml:15]: Otherwise cannot push as the HEAD is not attached to any branch

**Review by aakoshh — APPROVED** [review]
Nice one :+1:

File: .github/workflows/test.yml (hunk 1 above)
Comment (aakoshh) [reply .github/workflows/test.yml:15]: Thanks, it's worth adding a comment about this I reckon
